setup/src/main/java/io/jenkins/jenkinsfile/runner/JenkinsLauncher.java

```diff
@@ -77,7 +77,8 @@ protected ServletContext createWebServer() throws Exception {
                 ? new Server(launcherOptions.httpPort)
                 : new Server(queuedThreadPool);
 
-        WebAppContext context = new WebAppContext(launcherOptions.warDir.getPath(), null);
+        contextPath = launcherOptions.httpPath;
```

Comment:
You may need to handle mistakes here:

* trailing `/`
* empty string
* just `/`, which `WebAppContext` will like but `getURL` will not

Reply:
what do you want me to do with all of those. Just throw an Illegal Argument Exception? I agree that probably YAGNI so I don't want to waste too much time with error cases for something that'll probably not get used

Reply:
> Just throw an Illegal Argument Exception?

Sure, just to make things more transparent.

> I don't want to waste too much time with error cases for something that'll probably not get used

Agreed.

Reply:
done. Case 1 and 3 are covered by the same "ends with /" statement